### DISCUSSION TITLE
Running test code after every page load

### DISCUSSION THREAD

**Discussion by mateuszdrwal** [cypress-io/cypress#17162] [Questions and Help]
Hi, we are testing a PHP app using cypress and would like to make cypress fail the current test if it detects a PHP error on the page. Because we are using xdebug all PHP errors contain the class `.xdebug-error`, so detecting the error itself is as simple as `cy.get('.xdebug-error').should('not.exist')`. The problem is that with this solution we have to place this code after every cy.visit, form submit etc. which is easy to forget and reduces the code quality of the tests. Ideally the check would run automatically after every page load. We have tried running the following code on cypress init, but it throws [this error](https://docs.cypress.io/guides/references/error-messages#Cypress-detected-that-you-returned-a-promise-from-a-command-while-also-invoking-one-or-more-cy-commands-in-that-promise) with a not very helpful (in this case) message:

```js
Cypress.on('window:load', (window) => {
    cy.get('.xdebug-error').should('not.exist')
})
```

The same happens with the `onLoad` option to `cy.visit()`. I assume this is because you are not supposed to be able to queue asynchronous test code in events by running `cy` methods, or maybe I'm just doing it wrong. Even if the above code block is "completely wrong" with regard to how cypress works, I think it illustrates what we want to do pretty well. Is there some way to do that? Or maybe there is a better way to check for PHP errors in the first place without adding a bunch of test code after every page load?

**Comment by marmite22** [comment]
Did you ever find a solution to this? I need to do almost exactly the same thing

**Reply by mateuszdrwal** [reply]
Not really, what we did was to render PHP errors that throw JS errors that fail cypress

**Comment by mkonev-casafari** [comment]
Hello,
I think error message of problem says clearly what happening, so you need to check existence of element without 'Promise' like functions.

eg.
```
Cypress.Commands.overwrite('visit', (originalFn, args) => {
      originalFn(args).then(() => {
        expect(Cypress.$('.xdebug-error'), 'Error detected on page').length(0);
      });
    });
```

and then call your visit
```
cy.visit('url')
```

**Reply by mkonev-casafari** [reply]
@mateuszdrwal  @marmite22
